### PR TITLE
feat: support yaml to config plugin in plugin config page

### PR DESCRIPTION
**Pull request by liuxiran**
Please answer these questions before submitting a pull request

- Why submit this pull request?
- [ ] Bugfix
- [ ] New feature provided
- [ ] Improve performance
- [ ] Backport patches

- Related issues
 Part of #1039 

___
### New feature or improvement
- Describe the details and related test reports.

1 
image

2
image

Notice: all above are just transforms in FE, it would not disturb the interface data transfer 

___

### CODE REVIEW

File: web/src/components/Plugin/PluginDetail.tsx [hunk 1]
```diff
@@ -243,6 +268,7 @@ const PluginDetail: React.FC<Props> = ({
         </Form>
         <Divider orientation="left">Data Editor</Divider>
         <PageHeader
+          className="test"
```
Comment (bzp2010) [inline web/src/components/Plugin/PluginDetail.tsx:271]: Is the style sheet here added correctly?

File: web/src/components/Plugin/PluginDetail.tsx [hunk 2]
```diff
@@ -92,6 +93,11 @@ const PluginDetail: React.FC<Props> = ({
   const ref = useRef<any>(null);
   const data = initialData[name] || {};
   const pluginType = pluginList.find((item) => item.name === name)?.type;
+  const [codeMirrorMode, setCodeMirrorMode] = useState<PluginComponent.CodeMirrorMode>('javascript');
+  const modeOptions = [
+    { label: 'Json', value: 'javascript' },
```
Comment (LiteSun) [inline web/src/components/Plugin/PluginDetail.tsx:98]: why use `javascript ` here? 🤔

File: web/src/components/Plugin/PluginDetail.tsx (hunk 1 above)
Comment (liuxiran) [reply web/src/components/Plugin/PluginDetail.tsx]: > Is the style sheet here added correctly?

good catch, this style sheet is useless, fixed, thanks a lot

File: web/src/components/Plugin/PluginDetail.tsx (hunk 2 above)
Comment (liuxiran) [reply web/src/components/Plugin/PluginDetail.tsx]: > why use `javascript ` here? 🤔

thanks already changed to `json`

File: web/src/helpers.tsx [hunk 3]
```diff
@@ -130,3 +132,20 @@ export const timestampToLocaleString = (timestamp: number) => {
 
   return moment.unix(timestamp).format('YYYY-MM-DD HH:mm:ss');
 };
+
+/**
+ * Transform json string to yaml string
+ * @param jsonStr
+ */
+export const json2yaml = (jsonStr: string): string => {
+  return yaml.dump(JSON.parse(jsonStr))
```
Comment (LiteSun) [inline web/src/helpers.tsx:141]: if jsonStr's value is `''`, an error will occur

Comment (liuxiran) [reply web/src/helpers.tsx]: thanks a lot, already fixed.

use `try...catch` to handle the exception

**Comment by LiteSun** [comment]
Front-end e2e test required.

**Comment by liuxiran** [comment]
> Front-end e2e test required.

got it, I`ll add it soon, thanks

**Comment by juzhiyuan** [comment]
I have no idea how to resolve this conflict 😂 please fix it when you have time.

**Comment by membphis** [comment]
ping @LiteSun

**Review by LiteSun — APPROVED** [review]

**Review by juzhiyuan — APPROVED** [review]

**Pull request merged**